ovs/extensions/packages/packagefactory.py

```diff
@@ -29,36 +30,86 @@ class PackageFactory(_PackageFactory):
     """
     _logger = Logger('extensions-packages')
 
-    universal_packages = ['arakoon', 'openvstorage', 'openvstorage-backend', 'openvstorage-sdm']
-    ose_only_packages = ['alba', 'volumedriver-no-dedup-base', 'volumedriver-no-dedup-server']
-    ee_only_packages = ['alba-ee', 'volumedriver-ee-base', 'volumedriver-ee-server']
+    def __init__(self):
+        """
+        Initialization method
+        """
+        super(PackageFactory, self).__init__()
 
-    universal_binaries = ['arakoon']
-    ose_only_binaries = ['alba', 'volumedriver-no-dedup-server']
-    ee_only_binaries = ['alba-ee', 'volumedriver-ee-server']
+    @classmethod
+    def get_components(cls):
+        """
+        Retrieve the components which relate to this repository
+        :return: A set of components
+        :rtype: set
+        """
+        return {_PackageFactory.COMP_FWK, _PackageFactory.COMP_SD}
 
     @classmethod
-    def _get_packages(cls):
-        package_names = cls.ose_only_packages + cls.ee_only_packages
-        binaries = cls.ose_only_binaries + cls.ee_only_binaries
+    def get_package_info(cls):
+        """
+        Retrieve the package information related to the framework
+        This must return a dictionary with keys: 'names', 'edition', 'binaries', 'blocking', 'version_commands' and 'mutually_exclusive'
+            Names: These are the names of the packages split up per component related to this repository (framework)
+                * Framework
+                    * PKG_ARAKOON            --> Used for arakoon-config cluster and arakoon-ovsdb cluster
+                    * PKG_OVS                --> Code itself ... duh
+                    * PKG_OVS_EXTENSIONS     --> Extensions code is used by the framework repository
+                * StorageDriver
+                    * PKG_ARAKOON            --> StorageDrivers make use of the arakoon-voldrv cluster
+                    * PKG_VOLDRV_BASE(_EE)   --> Code for StorageDriver itself
+                    * PKG_VOLDRV_SERVER(_EE) --> Code for StorageDriver itself
+            Edition: Used for different purposes
+            Binaries: The names of the packages that come with a binary (also split up per component)
+            Blocking: Update will stop (block) in case the update fails for any of these packages
+            Version_commands: The commandos used to determine which binary version is currently active
+            Mutually_exclusive: Packages which are not allowed to be installed depending on the edition. Eg: ALBA_EE cannot be installed on a 'community' edition
+        :raises ValueError: * When configuration management key 'ovs/framework/edition' does not exists
+                            * When 'ovs/framework/edition' does not contain 'community' or 'enterprise'
+        :return: A dictionary containing information about the expected packages to be installed
+        :rtype: dict
+        """
+        edition_key = '/ovs/framework/edition'
```

Comment:
Might be useful to move all these the majority of these keys to methods/properties on Configuration

Reply:
Will be taken care of in https://github.com/openvstorage/framework/issues/1876